### PR TITLE
[v0.26.0rc][BugFix][Triton] Guard rejection sampler predecessor loads

### PR DESCRIPTION
**Pull request by Dawn952**
### What this PR does / why we need it?

Backport of #14902 to `releases/v0.26.0rc`.

The rejection sampler computes each request's start offset from the previous
entry of a cumulative-token buffer. Four Triton kernels use
`tl.where(index == 0, 0, tl.load(ptr + index - 1))`.

`tl.where` evaluates both operands, so request zero still issues the
predecessor load from `ptr[-1]`. Whether this faults depends on the allocator
mapping immediately before the tensor, which can turn the bug into an
intermittent MTE DDR out-of-range error during startup kernel warmup or the
first speculative-decode request.

This patch masks predecessor loads directly in the greedy, random, expand,
and recovered-token kernels, supplies neutral values for masked lanes, and
guards expand stores for padded request lanes.

Triton documents that both `tl.where` operands are evaluated and recommends
using load/store masks to avoid unintended memory operations:
https://triton-lang.org/main/python-api/generated/triton.language.where.html

### Does this PR introduce _any_ user-facing change?

No API or model-semantic change. It prevents nondeterministic out-of-bounds
memory accesses in speculative-decoding rejection-sampler kernels.

### How was this patch tested?

- `python -m py_compile vllm_ascend/ops/triton/reject_sample.py`
- `TORCH_DEVICE_BACKEND_AUTOLOAD=0 python -m pytest -q tests/ut/model_executor/warmup/test_rejection_sampler_triton_warmup.py tests/ut/model_executor/warmup/test_kernel_warmup.py` (3 passed)
- `git diff --check origin/releases/v0.26.0rc...HEAD`
- Existing single-card NPU rejection-sampler tests exercise request zero in all affected kernels; NPU execution is left to CI.

- vLLM version: v0.26.0
- vLLM main: https://github.com/vllm-project/vllm/commit/d02df748bf9efd99022f1a062597dc3cb3808485

### CODE REVIEW

File: vllm_ascend/ops/triton/reject_sample.py [hunk 1]
```diff
@@ -108,8 +108,10 @@ def rejection_greedy_sample_triton(
         is_greedy = tl.load(is_greedy_ptr + offset, mask=mask, other=0)
         is_greedy_mask = mask & (is_greedy != 0)
 
-    start_idx = tl.where(offset == 0, 0, tl.load(cu_num_draft_tokens_ptr + offset - 1, is_greedy_mask))
-    end_idx = tl.load(cu_num_draft_tokens_ptr + offset, is_greedy_mask)
+    prev_mask = is_greedy_mask & (offset > 0)
+    prev_end_idx = tl.load(cu_num_draft_tokens_ptr + offset - 1, mask=prev_mask, other=0)
+    start_idx = tl.where(offset == 0, 0, prev_end_idx)
+    end_idx = tl.load(cu_num_draft_tokens_ptr + offset, mask=is_greedy_mask, other=0)
     num_draft_tokens = end_idx - start_idx
```
Comment (Ronald1995) [inline vllm_ascend/ops/triton/reject_sample.py:115]: please add a test case for this corner case

File: vllm_ascend/ops/triton/reject_sample.py [hunk 2]
```diff
@@ -108,8 +108,10 @@ def rejection_greedy_sample_triton(
         is_greedy = tl.load(is_greedy_ptr + offset, mask=mask, other=0)
         is_greedy_mask = mask & (is_greedy != 0)
 
-    start_idx = tl.where(offset == 0, 0, tl.load(cu_num_draft_tokens_ptr + offset - 1, is_greedy_mask))
-    end_idx = tl.load(cu_num_draft_tokens_ptr + offset, is_greedy_mask)
+    prev_mask = is_greedy_mask & (offset > 0)
+    prev_end_idx = tl.load(cu_num_draft_tokens_ptr + offset - 1, mask=prev_mask, other=0)
```
Comment (AuroraEmiya) [inline vllm_ascend/ops/triton/reject_sample.py]: start_idx = tl.where(offset == 0, 0, tl.load(cu_num_draft_tokens_ptr + offset - 1, is_greedy_mask&(offset>0))
would be clean and readable.

File: vllm_ascend/ops/triton/reject_sample.py (hunk 1 above)
Comment (AuroraEmiya) [reply vllm_ascend/ops/triton/reject_sample.py]: The same and if this ops dose not have markdown file now . Add it in vllm_ascend/ops/triton/docs

File: vllm_ascend/ops/triton/reject_sample.py [hunk 3]
```diff
@@ -203,8 +205,10 @@ def rejection_random_sample_kernel(
     mask = offsets < vec_len
     is_greedy = tl.load(is_greedy_ptr + offsets, mask, other=1)
     not_greedy_mask = is_greedy == 0
-    start_idxs = tl.where(offsets == 0, 0, tl.load(cu_num_draft_tokens_ptr + offsets - 1, not_greedy_mask))
-    end_idxs = tl.load(cu_num_draft_tokens_ptr + offsets, not_greedy_mask)
+    prev_mask = not_greedy_mask & (offsets > 0)
+    prev_end_idxs = tl.load(cu_num_draft_tokens_ptr + offsets - 1, mask=prev_mask, other=0)
+    start_idxs = tl.where(offsets == 0, 0, prev_end_idxs)
```
Comment (AuroraEmiya) [inline vllm_ascend/ops/triton/reject_sample.py]: The same logic as above and the below should be reviewed as well

File: vllm_ascend/ops/triton/reject_sample.py [hunk 4]
```diff
@@ -108,8 +108,10 @@ def rejection_greedy_sample_triton(
         is_greedy = tl.load(is_greedy_ptr + offset, mask=mask, other=0)
         is_greedy_mask = mask & (is_greedy != 0)
 
-    start_idx = tl.where(offset == 0, 0, tl.load(cu_num_draft_tokens_ptr + offset - 1, is_greedy_mask))
```
Comment (AuroraEmiya) [inline vllm_ascend/ops/triton/reject_sample.py:111]: start_idx = tl.where(offset == 0, 0, tl.load(cu_num_draft_tokens_ptr + offset - 1, is_greedy_mask&(offset>0))
would be clean and readable. And so as below

File: vllm_ascend/ops/triton/reject_sample.py
Comment (AuroraEmiya) [inline vllm_ascend/ops/triton/reject_sample.py:1]: Add a markdown file under ‎vllm_ascend/ops/triton/docs
